### ISSUE TITLE
Doesn't include all chunks when `splitChunks` has the `name` is `false`

### ISSUE THREAD

**Issue by evilebottnawi** [jantimon/html-webpack-plugin#1347]
## Expected behaviour

Should works

## Current behaviour

Doesn't work

## Environment

Tell us which operating system you are using, as well as which versions of Node.js, npm, webpack, and html-webpack-plugin. Run the following to get it quickly:

```
Node.js v10.15.2
linux 5.3.0-23-generic
6.13.7
test@1.0.0 /home/evilebottnawi/IdeaProjects/webpack-bug-report
└── webpack@4.41.6 
```

### Config

Copy the minimal `webpack.config.js` to produce this issue:

```js
const HtmlWebpackPlugin = require('html-webpack-plugin');
const { CleanWebpackPlugin } = require('clean-webpack-plugin');
const webpack = require('webpack');
const { resolve } = require('path');

const NODE_ENV = process.env.NODE_ENV;

const isEnvDev = NODE_ENV === 'development';
const isEnvProd = NODE_ENV === 'production';

module.exports = {
  entry: './src/index.js',
  output: {
    path: isEnvProd ? resolve(__dirname, '../build') : undefined,
    filename: isEnvProd
      ? 'static/js/[name].[contenthash:10].js'
      : 'static/js/bundle.js',
    chunkFilename: isEnvProd
      ? 'static/js/[name].[contenthash:10].chunk.js'
      : 'static/js/[name].chunk.js',
    publicPath: '/'
  },
  plugins: [
    new HtmlWebpackPlugin({
      template: './public/index.html',
      minify: isEnvProd
        ? { collapseWhitespace: true, removeComments: true }
        : undefined
    }),
    isEnvProd && new CleanWebpackPlugin(),
    isEnvDev && new webpack.HotModuleReplacementPlugin()
  ].filter(Boolean),
  mode: NODE_ENV,
  devServer: {
    compress: true,
    contentBase: resolve(__dirname, '../dist'),
    hot: true,
    host: 'localhost',
    port: 3000,
    open: true
  },
  optimization: {
    splitChunks: {
      chunks: 'all',
      name: false
    }
  }
};
```

Copy your template file if it is part of this issue:

```html
<!DOCTYPE html>
<html lang="en">
  <meta charset="UTF-8">
  <meta name="viewport" content="width=device-width, initial-scale=1.0">
  <title>webpack</title>
</head>
<body>
  <h1>hello webpack</h1>
</body>
</html>
```

## Relevant Links

Original issue: https://github.com/webpack/webpack/issues/10419
Repo: https://github.com/xpromise/webpack-bug-report

- If your project is public, link to the repo so we can investigate directly.
- **BONUS POINTS:** Create a [minimal reproduction](http://stackoverflow.com/help/mcve) and upload it to GitHub. This will get you the fastest support.

## Additional context

No

**Comment by jantimon** [comment]
Which version?

**Comment by evilebottnawi** [comment]
@jantimon latest - `"html-webpack-plugin": "^3.2.0",`

**Comment by evilebottnawi** [comment]
Just try to run reproducible test repo

**Comment by jantimon** [comment]
@evilebottnawi can't reproduce or I don't understand the issue.

As you can see `npm run build`  does generate only 1 chunk which is included:

![bug](https://user-images.githubusercontent.com/4113649/75029800-09db7080-54a3-11ea-8383-1ead1347470d.png)

**Comment by evilebottnawi** [comment]
@jantimon `npm run start`

**Comment by evilebottnawi** [comment]
With `name: false`:
```html
<!DOCTYPE html>
<html lang="en">
  <meta charset="UTF-8">
  <meta name="viewport" content="width=device-width, initial-scale=1.0">
  <title>webpack</title>
</head>
<body>
  <h1>hello webpack</h1>
<script type="text/javascript" src="/static/js/bundle.js"></script></body>
</html>
```

`0.chunk.js` is missing

With commented `name: false`:

```html
<!DOCTYPE html>
<html lang="en">
  <meta charset="UTF-8">
  <meta name="viewport" content="width=device-width, initial-scale=1.0">
  <title>webpack</title>
</head>
<body>
  <h1>hello webpack</h1>
<script type="text/javascript" src="/static/js/vendors~main.chunk.js"></script><script type="text/javascript" src="/static/js/bundle.js"></script></body>
</html>
```

**Comment by jantimon** [comment]
fixed with html-webpack-plugin 4.0.1
